src/rtracer.js

```diff
@@ -221,7 +221,11 @@ const hapiPlugin = ({
 
     if (echoHeader) {
       server.ext('onPreResponse', async (request, h) => {
-        request.response.header(headerName, id())
+        if (request.response && request.response.header) {
+          request.response.header(headerName, id())
+        } else if (request.response && request.response.output && request.response.output.headers) {
+          request.response.output.headers[headerName] = id()
+        }
```

Comment:
you can also avoid all this and do: 
```suggestion
        request.raw.res.setHeader(headerName, id())
```
which I kinda prefer and passes all the tests. I kinda prefer this way better but you end up crossing the Hapi/NodeJS barrier.

Reply:
Resolving this one in favor of https://github.com/puzpuzpuz/cls-rtracer/pull/45#discussion_r524931696